scripts/distro-test/amzn/Dockerfile

```diff
@@ -1,5 +1,6 @@
 FROM amazonlinux:latest
 
-RUN yum -y install curl sudo
+RUN yum -y install sudo
+RUN if ! [ -x "$(command -v curl)" ]; then yum -y install curl; exit 1; fi
```

Comment:
What's the purpose of the `exit 1` here?

Also, detecting if curl exists can be done by checking the status code of `command`, e.g.
```
if ! command -v curl >/dev/null; then printf 'curl is not installed'; fi
```